### PR TITLE
Splitting into pre and post proguard tests

### PR DESCRIPTION
**Pull request by ashwinraghav**
This change makes it possible to run the smoke tests on unproguarded versions of apps.
The goal is to have smoke tests that run on unproguarded apps and on proguarded apps.

Given the way the android plugin works, it seems hard to do this in the same gradle run, and may require that we have two separate gradle invocations for these tests. 

```
./gradlew connectedCheck -PtestBuildType=release -PfirebaseProjectId=<your_project_id> -PfirebaseToken=<your_firebase_token> -Pm2Repository=${PWD}/../build/m2repository/
```
and
```
./gradlew connectedCheck -PtestBuildType=debug -PfirebaseProjectId=<your_project_id> -PfirebaseToken=<your_firebase_token> -Pm2Repository=${PWD}/../build/m2repository/
```

### CODE REVIEW

**Comment by ashwinraghav** [comment]
@vkryachko Just confirming that we are yet to introduce the maven cache proxy.
https://k8s-gubernator.appspot.com/build/android-ci/pr-logs/pull/firebase_firebase-android-sdk/36/connected-check/1043149508671705090/

**Comment by vkryachko** [comment]
The proxy is enabled, we have not yet disabled the local shared mount of modules-2, will do today

File: ci/fireci/fireci/commands.py [hunk 1]
```diff
@@ -26,19 +26,34 @@
...
-  gradle.run('publishAllToBuildDir')
-
-  cwd = os.getcwd()
-  gradle.run(
-      'connectedReleaseAndroidTest',
-      gradle_opts='-Dmaven.repo.local={}'.format(
-          os.path.join(cwd, 'build', 'm2repository')),
-      workdir=os.path.join(cwd, 'test-apps'),
-  )
+def smoke_tests_proguarded_app():
```
Comment (vkryachko) [inline ci/fireci/fireci/commands.py]: both tasks look almost identical, did you consider having a single parameterized `smoke_tests` command with a `@click.option('--build-type')`?

**Pull request merged**